cherrypick-1.1: util/log: parse logs assuming UTC

dt opened the pull request:
72d005482 switched `formatLogEntry` to use `timeutil.Unix` and thus changed the `time.Time` passed to `formatHeader` to have a UTC location, meaning the log timestamps switched to being printed in UTC rather than the host's local TZ.

This is actually likely desireable in a distributed system, as it makes correlating events across nodes in a geographically diverse cluster easier, but it broke our parsing of the logs, which assumed local.

We probably shouldn't assume any particular offset anyway when reading a log file that might have been written with a different offset. We could add explicit offsets, to avoid assuming anything, but as mentioned above, just switching to UTC everywhere has other benefits too.# Please enter the commit message for your changes. Lines starting

dt committed:
util/log: parse logs assuming UTC

72d005482 switched `formatLogEntry` to use `timeutil.Unix` and thus changed the `time.Time` passed to `formatHeader` to have a UTC location, meaning the log timestamps switched to being printed in UTC rather than the host's local TZ.

This is actually likely desireable in a distributed system, as it makes correlating events across nodes in a geographically diverse cluster easier, but it broke our parsing of the logs, which assumed local.

We probably shouldn't assume any particular offset anyway when reading a log file that might have been written with a different offset. We could add explicit offsets, to avoid assuming anything, but as mentioned above, just switching to UTC everywhere has other benefits too.# Please enter the commit message for your changes. Lines starting

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/18867)
<!-- Reviewable:end -->

dt commented:
er, might need another one in here, waiting for now.

dt committed:
util/log: use UTC in log tests too

We had previously whitelisted a few logging-related files from being required to use timeutil.
Requiring them to do so as well ensures they too use the helpers that return UTC rather than Local.

dt commented:
okay, added #18868 too.

petermattis reviewed (approved):
LGTM
